packages/smooth_app/lib/pages/preferences/country_selector/country_selector_provider.dart

```diff
@@ -47,146 +50,56 @@ class _CountrySelectorProvider extends PreferencesSelectorProvider<Country> {
-          Country(
-              name: _fixCountryName(countryName),
-              countryCode: _fixCountryCode(countryCode)),
-        );
-        continue;
-      }
-      final String fixedCountryCode = _fixCountryCode(countryCode);
-      final Country country = fixedCountryCode == countryCode
-          ? localizedCountry
-          : Country(name: localizedCountry.name, countryCode: countryCode);
-      finalCountriesList.add(country);
-    }
-
-    return finalCountriesList;
-  }
-
-  /// Fix the countryCode if needed so Backend can process it.
-  static String _fixCountryCode(String countryCode) {
-    // 'gb' is handled as 'uk' in the backend.
-    if (countryCode == 'gb') {
-      countryCode = 'uk';
-    }
-    return countryCode;
-  }
-
-  /// Fix the issues where United Kingdom appears with lowercase 'k'.
-  static String _fixCountryName(String countryName) {
-    if (countryName == 'United kingdom') {
-      countryName = 'United Kingdom';
-    }
-    return countryName;
-  }
-
-  /// Reorder countries alphabetically, bring user's locale country to top.
   static void _reorderCountries(
-    List<Country> countries,
+    List<OpenFoodFactsCountry> countries,
     String? userCountryCode,
   ) {
+    final Map<OpenFoodFactsCountry, LocalizedCountry> localizedMap = {
+      for (final c in LocalizedCountry.getLocalizedCountries()) c.country: c
+    };
+
     countries.sort(
-      (final Country a, final Country b) {
-        if (a.countryCode == userCountryCode) {
+      (a, b) {
+        if (a.offTag == userCountryCode) {
           return -1;
         }
-        if (b.countryCode == userCountryCode) {
+        if (b.offTag == userCountryCode) {
           return 1;
         }
-        return a.name.compareTo(b.name);
+        final String nameA = localizedMap[a]?.localizedName ?? '';
+        final String nameB = localizedMap[b]?.localizedName ?? '';
+        return nameA.compareTo(nameB);
       },
     );
   }
 
   @override
-  Country getSelectedValue(List<Country> countries) {
+  OpenFoodFactsCountry getSelectedValue(List<OpenFoodFactsCountry> countries) {
     if (userCountryCode != null) {
-      for (final Country country in countries) {
-        if (country.countryCode.toLowerCase() ==
-            userCountryCode?.toLowerCase()) {
-          return country;
-        }
-      }
+      return countries.firstWhere(
+        (OpenFoodFactsCountry country) =>
+            country.offTag.toLowerCase() == userCountryCode?.toLowerCase(),
+        orElse: () => countries.first,
+      );
     }
-    return countries[0];
+    return countries.first;
   }
 
   @override
-  Future<void> onSaveItem(Country country) => preferences.setUserCountryCode(
-        country.countryCode,
-      );
-}
-
-class CountriesHelper {
-  const CountriesHelper._();
+  Future<void> onSaveItem(OpenFoodFactsCountry country) =>
+      preferences.setUserCountryCode(
+          country.offTag); // Save offTag (not iso2code) in preferences
```

Comment:
With your new code we slightly change the value (from ISO2CODE to offTag, including "GB" / "uk") but that shouldn't matter that much as `OpenFoodFactsCountry.fromOffTag` is quite resilient.
Eventually remove the comment: let's pretend we always used offTag.